p2p/net/swarm/swarm_dial.go

```diff
@@ -44,6 +44,9 @@ var (
 // add loop back in Dial(.)
 const dialAttempts = 1
 
+// dial at most 12 peers concurrently
+const concurrentPeerDials = 12
```

Comment:
and, `12` max concurrent dials is a bit rough. Maybe this should only be a limit for socket-consuming transports?